SwEnt-2024-Group-19/Activities
# Feature/activities/expiricy time

This pull request introduces several enhancements and bug fixes related to time handling, filtering logic, and profile management. It also includes updates to test coverage and dummy data adjustments to ensure accuracy.
### Time Management:
- Added a new function to seamlessly combine timestamps with hour strings for enhanced time operations.
- Centralized time handling logic for reusability across components.
### Overview Screen:
- Past activities are now effectively excluded, improving the user experience by focusing on upcoming and ongoing activities.
- Corresponding tests ensure robustness of this filtering mechanism.
### Testing:
- Enhanced test coverage with targeted tests for filtering logic, improving the reliability of the changes.
### How to test?
1. Create an activity that will start in the next 5 minutes
2. Check in the overview screen that the activty will disappear
### OR
1. Create an activity that will start in the next 5 minutes
2. Check in the profile screen that the activity moves from the "created" box to the "past" box

- [x] Added new tests for the time filtering logic.
- [x] Refactored profile filtering logic without introducing breaking changes.
- [x] Updated dummy data to fix incorrect fields.
- [x] Verified the overview screen does not display past activities if they are still not past.
